### PR TITLE
Add utility functions to work with node groups

### PR DESCRIPTION
**Pull request by elpoelma**
### Overview
This PR adds two new utility functions:
- `getGroups(node: PNode): string[]`: this function returns the list of groups a given node belongs to
- `hasGroups(node: PNode, ...groups: string[]): boolean`: this function checks whether a given node belongs to a given series of groups.

These utility functions can be useful in order to implement functionality which is applicable to a specific node-group.

##### connected issues and PRs:
None

### Checks PR readiness
- [x] changelog
- [x] npm lint
- [x] no new deprecations

### CODE REVIEW

File: addon/utils/_private/node-utils.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { PNode } from '@lblod/ember-rdfa-editor';
+
+export function getGroups(node: PNode) {
+  return node.type.spec.group?.split(' ') ?? [];
+}
+
+export function hasGroups(node: PNode, ...groups: string[]) {
+  const nodeGroups = getGroups(node);
+  return groups.every((group) => nodeGroups.includes(group));
+}
```
Comment (x-m-el) [inline addon/utils/_private/node-utils.ts:10]: Might be an idea to also support `hasGroup`?
Or support passing just a string too (but then the function name makes less sense and that's probably not javascripty).
As I'm guessing you'd more often check for one specific group than a list of them.

**Review by x-m-el — APPROVED** [review]
small comment, reset looks good :+1: 
(I didn't test it)

File: addon/utils/_private/node-utils.ts (hunk 1 above)
Comment (elpoelma) [reply addon/utils/_private/node-utils.ts:10]: yes but that function supports that. You can invoke it by using `hasGroups(node, 'group')` or `hasGroups(node, 'group1', 'group2')`

Comment (x-m-el) [reply addon/utils/_private/node-utils.ts:10]: Great, I looked over the spread operator :+1:

**Pull request merged**